### PR TITLE
fix(learn): Fix the public assets folder confusion by giving it its own folder

### PR DESCRIPTION
**Pull request by ehsabd**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Although it is mentioned in the description that an HTML server usually has one or more public assets directories, the assets had been mounted to the root folder. I suppose this is non-standard and confusing. 

I understand perhaps the author(s) wanted this to work with the simple idea of having a `/style.css` file in the `index.html` file in the boilerplate, but the `index.html` view should align with a node server best practices not the other way around.

- Changed the instruction so that the code mounts `public` folder to `public` path instead of root.
- Added a note about how this relates to the `index.html` file.
- NOTE that this works paired with this PR in freeCodeCamp/boilerplate-express : freeCodeCamp/boilerplate-express#51

### CODE REVIEW

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md:19]:
```suggestion
Now your app should be able to serve a CSS stylesheet. Note that the `/public/style.css` file is referenced in the `/views/index.html` in the project boilerplate. Your front-page should look a little better now!
```
This note can be removed, IMHO.

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md:23]:
```suggestion
Your app should serve asset files from the `/public` directory to the `/public` path
```

**Review by naomi-lgbt — CHANGES_REQUESTED** [review]

**Comment by ehsabd** [comment]
@nhcarrigan Can you also review this PR: https://github.com/freeCodeCamp/boilerplate-express/pull/51. Because it's related to this one. They only work together.

File: curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md
Comment (nhcarrigan) [inline curriculum/challenges/english/05-apis-and-microservices/basic-node-and-express/serve-static-assets.md:13]:
```suggestion
An HTML server usually has one or more directories that are accessible by the user. You can place there the static assets needed by your application (stylesheets, scripts, images).

In Express, you can put in place this functionality using the middleware `express.static(path)`, where the `path` parameter is the absolute path of the folder containing the assets.

If you don’t know what middleware is... don’t worry, we will discuss in detail later. Basically, middleware are functions that intercept route handlers, adding some kind of information. A middleware needs to be mounted using the method `app.use(path, middlewareFunction)`. The first `path` argument is optional. If you don’t pass it, the middleware will be executed for all requests.
```

Were these intended to be rendered as separate lines? If so, you'll need to add a blank new-line between them.

**Review by naomi-lgbt — APPROVED** [review]
Looks good to me.

Tested locally with the related changes to the boilerplate, no issue getting tests to pass.

Thank you for all of your work on the changes. Please note that our workflow requires at least one additional approval before this can be merged.

**Comment by ehsabd** [comment]
@raisedadead  please also review and merge this PR: freeCodeCamp/boilerplate-express#51. Because it's related to the current one. They only work together.

**Comment by raisedadead** [comment]
Thanks for bringing it to my attention. I just merged the boilerplate PR. Thanks for contributing.

**Comment by ShaunSHamilton** [comment]
@ehsabd Just for info, boilerplate PRs should not be merged, until corresponding `/learn` PRs are merged into the `prod-current` branch - this is tied with the CI pipeline, whereas every commit merged into one of the boilerplate repos goes directly into the branch used in _'production'_.

The reason for this is to prevent the case where the boilerplate is ahead of the instructions/tests on the main platform.

Right now, things are moving quite fast, but sometimes it takes a while for `/learn` commits to end up in production.

**Pull request merged**
